neptune/new/internal/backends/hosted_neptune_backend.py

```diff
@@ -207,12 +207,20 @@ def get_project(self, project_id: str) -> Project:
                 raise NeptuneLegacyProjectException(project_id)
             return Project(project.id, project.name, project.organizationName)
         except HTTPNotFound:
+            available_workspaces = self.get_available_workspaces()
+
+            if workspace and not any(
+                filter(lambda aw: aw.name == workspace, available_workspaces)
```

Comment:
`any()` takes an iterable of booleans, doesn't it? So i think it should be either:
```any(map(lambda aw: aw.name == workspace, available_workspaces)```
or
```list(filter(lambda aw: aw.name == workspace, available_workspaces)```

Your code is correct as long as `workspace` evaluates to True (is not empty and is not None). But it's weird to read.

Reply:
Ok, list was already used in a few lines above :wink:

Reply:
and how would you like `next(filter(lambda aw: aw.name == workspace, available_workspaces), None) is None`? :D

Reply:
Nah, looks like a Python 2.x code :P